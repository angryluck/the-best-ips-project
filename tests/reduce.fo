fun int incr(int a, int b) = a + b
fun int writeInt(int n) = write(n)

fun int reduce_case([int] a) =
  let r = reduce(incr, 0, a) in
  writeInt(r)

fun int main() =
  let tmp = reduce_case({1, 2, 3}) in          
  let tmp = reduce_case({10, 20}) in           
  let tmp = reduce_case({0, 5, 3}) in                 
  let tmp = reduce_case({5}) in                
  let tmp = reduce_case({4, 4, 4, 4}) in      
  let tmp = reduce_case({100, 200, 300}) in     
  let tmp = reduce_case({0, 0, 0}) in           
  let tmp = reduce_case({1, 1, 1, 1, 1}) in     
  let tmp = reduce_case({2, 4, 6, 8}) in        
  reduce_case({9, 5, 1})                       
