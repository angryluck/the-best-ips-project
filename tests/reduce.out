6
30
8
5
16
600
0
5
20
15
